### PR TITLE
Leave flat - closes #204

### LINKED ISSUE #204: Users can leave flats

### PR DESCRIPTION
**Pull request by JessicaNeary**
Users can now leave flats via a button on the flat page

### CODE REVIEW

File: src/client/actions/index.js [hunk 1]
```diff
@@ -315,3 +315,25 @@ export const ignoreJoinRequest = requestId => {
       })
   }
 }
+
+export const leaveFlat = (userId, flatId) => {
+  return dispatch => {
+    dispatch({
+      type: 'LEAVE_FLAT_PENDING'
+    })
+    return getAxios().delete(`/v1/flat/${flatId}/${userId}`)
```
Comment (davidchristie) [inline src/client/actions/index.js:347]: Please rename route to `/v1/flats/:flatId/:userId`

File: src/client/reducers/account.js [hunk 2]
```diff
@@ -11,6 +11,14 @@ export default (state = {}, action) => {
         )
       }
 
+    case 'LEAVE_FLAT_SUCCESS':
+      const flats = state.user.flats
```
Comment (davidchristie) [inline src/client/reducers/account.js]: Please use `let` instead of `const` here.

File: src/server/routes/v1.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ router.post('/users', (req, res) => {
 })
 
 // Routes under this middleware require a valid token to access
-router.use(jwtMiddleware)
+//router.use(jwtMiddleware)
```
Comment (davidchristie) [inline src/server/routes/v1.js]: Please put auth middleware back in.

File: src/server/routes/v1.js [hunk 4]
```diff
@@ -143,6 +143,19 @@ router.put('/notes/:id', (req, res) => {
     })
 })
 
+router.delete('/flat/:flatId/:userId', (req, res) => {
```
Comment (davidchristie) [inline src/server/routes/v1.js:190]: Please rename this route to `/flats/:flatId/:userId`

**Review by davidchristie — CHANGES_REQUESTED** [review]

**Comment by AnnaUlyanova** [comment]
the changes will be applied with the branch aboutPageContent

**Pull request merged**